inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.cpp

```diff
@@ -0,0 +1,538 @@
+    if (endDims.size() > 1)
+        THROW_ERROR << "should have end vector with 1 dimension";
+    if (beginDims[0] != endDims[0])
+        THROW_ERROR << "should have begin vector with size equal to end vector size";
+
+    SizeVector strideDims;
+    if (stridedSliceLayer->insData.size() > STRIDE_ID) {
+        auto strideData = stridedSliceLayer->insData[STRIDE_ID].lock();
+        if (!strideData)
+            THROW_ERROR << "has nullable input data";
+
+        strideDims = strideData->getTensorDesc().getDims();
+        if (strideDims.size() > 1)
+            THROW_ERROR << "should have stride vector with 1 dimension";
+        if (beginDims[0] != strideDims[0])
+            THROW_ERROR << "should have stride vector with size equal to begin vector size";
+    }
+
+    auto createMask = [&](const char* maskName, std::vector<int>& mask, const int bit) {
+        std::string::size_type i;
+        std::string maskStr = stridedSliceLayer->GetParamAsString(maskName, "");
+        for (i = 0; i < maskStr.size(); ++i) {
+            if (maskStr[i] == '1')
+                mask.push_back(1);
+            else if (maskStr[i] == '0')
+                mask.push_back(0);
+            else if (maskStr[i] != ',')
+                THROW_ERROR << "has incorrect '" << maskName << "'. Mask values should be 0 or 1";
+        }
+        if (strcmp(maskName, "ellipsis_mask") != 0 || i == 0) {
+            for (; i < srcDims.size(); ++i) mask.push_back(bit);
+        }
+    };
+
+    createMask("begin_mask", beginMask, 1);
+    createMask("end_mask", endMask, 1);
+    createMask("new_axis_mask", newAxisMask, 0);
+    createMask("shrink_axis_mask", shrinkAxisMask, 0);
+    createMask("ellipsis_mask", ellipsisMask, 0);
+
+    int ellipsisMaskCounter = 0;
+    int ellipsisPos1 = -1;
+    for (size_t i = 0; i < ellipsisMask.size(); i++) {
+        ellipsisMaskCounter += ellipsisMask[i];
+        ellipsisPos1 = ellipsisMask[i] == 1 && ellipsisPos1 == -1 ? i : ellipsisPos1;
+    }
+    if (ellipsisMaskCounter > 1)
+        THROW_ERROR << "has incorrect 'Ellipsis_mask'. Only one non-zero bit is allowed";
+
+    int newAxis = 0;
+    for (auto na : newAxisMask)
+        newAxis += na;
+    size_t maxDims = srcDims.size() + newAxis;
+
+    int shrinkAxis = 0;
+    for (auto sa : shrinkAxisMask)
+        shrinkAxis += sa;
+    params.equalDims = srcDims.size() == maxDims && shrinkAxis == 0;
+
+    auto fillingInParameters = [&](std::vector<int>& parameter, const size_t type, const size_t size, const int bit) {
+        auto blob = getParentEdgesAtPort(type)[0]->getParent()->getCnnLayer()->blobs["custom"];
+        const int* ptr = blob->cbuffer().as<const int*>() + blob->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        parameter.assign(ptr, ptr + size);
+
+        if (ellipsisMaskCounter == 0 && size < dstDims.size()) {
+            for (size_t i = size; i < dstDims.size(); i++)
+                parameter.push_back(bit);
```

Comment:
No, because we have all masks of the same size (in your case will be bit adding in first position - beginMask[1] = endMask[1] = 1).
This bit adding in mask end was in last version StridedSlice. May be there is some model with StridedSlice which has mask with size less dstDims size (Not sure about it but I think should to stay it)

Reply:
UPD: The model from ONNX  test `test_onnx_model_zoo_vision_object_detection_segmentation_ssd_model_ssd_10_model_cpu` has masks and parameters with size less dstDims size. I added [test](https://github.com/openvinotoolkit/openvino/blob/42f372b6fb1b8c066abc300c28e5f59b77709f92/inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/strided_slice.cpp#L73) to cover this case